shell/src/main/java/tachyon/shell/TfsShell.java

```diff
@@ -174,6 +178,61 @@ public void load(TachyonURI filePath) throws IOException {
   }
 
   /**
+   * Persist a file or directory currently stored only in Tachyon to the UnderFileSystem
+   *
+   * @param filePath the TachyonURI path to persist to the UnderFileSystem
+   * @throws IOException
+   */
+  public void persist(TachyonURI filePath) throws IOException {
+    try {
+      TachyonFile fd = mTfs.open(filePath);
+      FileInfo fInfo = mTfs.getInfo(fd);
+      if (fInfo.isFolder) {
+        List<FileInfo> files = mTfs.listStatus(fd);
+        List<String> errorMessages = new ArrayList<String>();
+        for (FileInfo file : files) {
+          TachyonURI newPath = new TachyonURI(file.getPath());
+          try {
+            persist(newPath);
+          } catch (IOException e) {
+            errorMessages.add(e.getMessage());
+          }
+        }
+        if (errorMessages.size() != 0) {
+          throw new IOException(Joiner.on('\n').join(errorMessages));
+        }
+      } else if (fInfo.isIsPersisted()) {
+        System.out.println(filePath + " is already persisted");
+      } else {
+        Closer closer = Closer.create();
+        try {
+          InStreamOptions inStreamOptions = new InStreamOptions.Builder(mTachyonConf)
+              .setTachyonStorageType(TachyonStorageType.NO_STORE).build();
+          FileInStream in = closer.register(mTfs.getInStream(fd, inStreamOptions));
+
+          TachyonURI dstPath = new TachyonURI(fInfo.getUfsPath());
+          UnderFileSystem ufs = UnderFileSystem.get(dstPath.getPath(), mTachyonConf);
+          String parentPath = dstPath.getParent().getPath();
+          if (!ufs.exists(parentPath) && !ufs.mkdirs(parentPath, true)) {
```

Comment:
Since it's a ufs operation, I think we have to do it manually. Seems like the same thing is done in `FileSystemMaster`